services/vision/vision.go

```diff
@@ -118,10 +119,26 @@ func FindFirstName(r robot.Robot) string {
 	return ""
 }
 
-// FirstFromRobot returns the first vision service in this robot.
+// FirstFromRobot returns the first vision service on this robot.
 func FirstFromRobot(r robot.Robot) (Service, error) {
-	name := FindFirstName(r)
-	return FromRobot(r, name)
+	vis, err := FirstFromLocalRobot(r)
+	if err != nil {
+		name := FindFirstName(r)
+		return FromRobot(r, name)
+	}
+	return vis, err
+}
+
+// FirstFromLocalRobot returns the first vision service on this main robot.
+// This will specifically ignore remote resources.
+func FirstFromLocalRobot(r robot.Robot) (Service, error) {
+	for _, n := range r.ResourceNames() {
+		if n.Subtype == Subtype && !n.ContainsRemoteNames() {
+			service, err := r.ResourceByName(n)
+			return service.(Service), err
```

Comment:
If this isn't a (Service), it will panic. Even though it's unlikely it will, go ahead and wrap this in a 
```
if s, ok := service.(Service); ok {
   return s, nil
}
```

Reply:
would just do FromRobot at this point